src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -448,7 +458,10 @@ def process_parquet_queue_with_threads(
                 if fetch_with_process:
                     # Check if any processes have failed
```

Comment:
I think we need another check here to avoid inf loop:

if all thread futures re done, all processes are not alive, but len(completed_partitions) < total_partitions still holds, then it means some process has silently failed. we need to raise the error out